Recap of Compiler Research Organization’s participation in GSoD 2023 (#138)
compiler-research/compiler-research.github.io: _pages/blogs/GSoD23_QuillPusher.md

---
title: "Recap of Compiler Research Organization’s participation in Google’s Season of Docs 2023"
layout: gridlay
excerpt: "A review of the documentation efforts in 2023."
sitemap: false
permalink: blogs/gsod23_quillpusher_experience_blog/
---

# Recap of Compiler Research Organization’s participation in Google’s Season of Docs 2023

## How we got started

> An audit of the existing documentation was done  to identify the gaps and

Reviewer: ```suggestion
An audit of the existing documentation was done to identify gaps and
```

potential areas of improvement. Our main areas of focus were (these evolved
over time, based on documentation audit):

- Automatic Differentiation applications (using Clad, e.g. in RooFit and
  Floating-Point Error Estimation), and

- Python-C++ Interoperability (Clang-Repl (LLVM), CppInterOp, cppyy, Numba,
  etc.).

We also wanted to capture the [Student Success Stories], including milestones
achieved by student developers under the banner of Compiler Research
Organization’s guidance and mentorship programs.

## What we accomplished

Following are the major documentation areas that we worked on.

- AD Support in RooFit (ROOT)
- Execution Results Handling in Clang-REPL (LLVM)
- Floating Point Error Estimation (Clad)
- CppInterOp for Python and C++ Interoperability (Compiler Research
  Organization)
- cppyy Enhancements (Compiler Research Organization)

> - Numba Enhancements (cppyy)

Reviewer: It might be a good idea to make these links to the projects, blog, tutorials, etc.

Author: done

These were concluded based on expected timeline, except for some final reviews

> and un-merged PRs. A detailed report with relevant PRs can be found in the

Reviewer: ```suggestion
and yet to be merged PRs. A detailed report with relevant PRs can be found in the
```

Author: removing this line altogether, since all listed PRs are now merged

[GSoD 2023 Case Study].

## Our overall experience

We would consider the project successful, on account of the major documentation
contributions in upstream LLVM, ROOT, and other repositories, where our
developers had previously contributed code.

Going into this project, we had a vision of what we wanted to achieve, but over
several brainstorming sessions, we identified several areas that we thought
needed improvement. Due to the limited time frame that we were working with, we
prioritised the tasks based on what was urgent and what was important.

## GSoD contributor’s experience

> “I came into this project without much experience with compiler-related

Reviewer 2: why the "?

Reviewer: It’s a quote that ends 2 paragraphs below.

Author: The news item is written from the perspective of a general moderator, who interviewed me, and added my "comments" in the last two paragraphs :D

Reviewer 2: its quiet not obvious.. maybe you can add some introduction/glue to the article to say that and to make the transition to the quote clear? There is "we" and "I" so its not a "general moderator" but rather a team member...

technologies. I had technical writing experience, but in a different domain.
So, I had to shift gears and come in with a student’s mentality, while bringing
along the discipline of a seasoned writer.

My mentors (David Lange and Vassil Vassilev) were kind and patient as I learned
the ropes and started landing upstream documentation for large communities such

> as Root, LLVM, etc. Despite having years of experience before this, it was a

Reviewer: ```suggestion
as ROOT, LLVM, etc. Despite having years of experience before this, it was a
```
